### PR TITLE
Logs: Make logs container scrollable

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
makes the logs container scrollable

**Why do we need this feature?**
easily jump back to the query editor without having to scroll back through all of your logs

**Who is this feature for?**
...

**Which issue(s) does this PR fix?**:
Fixes #

**Special notes for your reviewer:**
Before & After:

https://github.com/grafana/grafana/assets/34524710/6a1b03ee-d904-4a27-b8c5-b0c3d18ac0ff

https://github.com/grafana/grafana/assets/34524710/8575c574-0ba7-4a5f-ae81-2753ff76b645

### CODE REVIEW

File: public/app/features/explore/Logs/Logs.tsx [hunk 1]
```diff
@@ -632,6 +633,8 @@ const getStyles = (theme: GrafanaTheme2, wrapLogMessage: boolean) => {
       display: flex;
       flex-direction: row;
       justify-content: space-between;
+      overflow-x: scroll;
+      height: calc(100vh - 150px);
```
[the remark this answers is not in the document]
Comment (gwdawson) [reply public/app/features/explore/Logs/Logs.tsx]: @svennergr this is just an estimated value that i found works to fit. it is needed to offset the height from the nav and some margin/padding from surrounding components

**Review by svennergr — COMMENTED** [review]
As per Slack convo there is a different behavior when using "Wrap Lines". I think that should be fixed somehow 😆

File: public/app/features/logs/components/getLogRowStyles.ts [hunk 2]
```diff
@@ -43,6 +44,8 @@ export const getLogLevelStyles = (theme: GrafanaTheme2, logLevel?: LogLevel) =>
 export const getLogRowStyles = memoizeOne((theme: GrafanaTheme2) => {
   const hoverBgColor = styleMixins.hoverColor(theme.colors.background.secondary, theme);
   const contextOutlineColor = tinycolor(theme.components.dashboard.background).setAlpha(0.7).toRgbString();
+  const scrollableLogsContainer = config.featureToggles.scrollableLogsContainer;
+  // const scrollableLogsContainer = false;
```
Comment (svennergr) [inline public/app/features/logs/components/getLogRowStyles.ts:48]:
```suggestion
```

**Review by svennergr — APPROVED** [review]
LGTM.

I think we should rethink the "Scroll to top" button and it's behaviour, but not necessarily in this PR.

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 3]
```diff
@@ -114,6 +114,7 @@ Experimental features might be changed or removed without prior notice.
 | `lokiPredefinedOperations`         | Adds predefined query operations to Loki query editor                                                        |
 | `pluginsFrontendSandbox`           | Enables the plugins frontend sandbox                                                                         |
 | `cloudWatchLogsMonacoEditor`       | Enables the Monaco editor for CloudWatch Logs queries                                                        |
+| `scrollableLogsContainer`          | Improves the scrolling behavior of logs in Explore                                                           |
```
Comment (ivanahuckova) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:117]:
```suggestion
| `exploreScrollableLogsContainer`          | Improves the scrolling behavior of logs in Explore                                                           |
```

**Review by ivanahuckova — COMMENTED** [review]
I would consider adding explore to the toggle name so it is clear that this won't affect any other logs container. 🙂

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md (hunk 3 above)
Comment (gwdawson) [reply docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:117]: this is valid, ill update it now 🙂

**Pull request merged**